composite/bitcoin-json-rpc/src/adapter.ts

```diff
@@ -0,0 +1,44 @@
+import JSONRPC from '@chainlink/json-rpc-adapter'
+import { Execute, AdapterRequest } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+import { Requester } from '@chainlink/external-adapter'
+
+const inputParams = {
+  url: false,
+  method: false,
+  params: false,
+  blockchain: false,
+  coin: false,
+  endpoint: false,
+}
+
+const convertEndpoint: { [key: string]: string } = {
+  height: 'headers',
+}
+
+// Export function to integrate with Chainlink node
+export const execute: Execute = async (request: AdapterRequest) => {
+  const validator = new Validator(request, inputParams)
+  const blockchain = validator.validated.data.blockchain || validator.validated.data.coin
+
+  let endpoint = validator.validated.data.endpoint
+
+  if (validator.error) throw validator.error
```

Comment:
We should throw the validation error right after validating right?
```suggestion
   const validator = new Validator(request, inputParams)
   if (validator.error) throw validator.error
  
  const blockchain = validator.validated.data.blockchain || validator.validated.data.coin
  let endpoint = validator.validated.data.endpoint
```

Reply:
True, changing this.